common/network_manager_utils.go

```diff
@@ -0,0 +1,76 @@
+package common
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	k6lib "go.k6.io/k6/lib"
+	k6netext "go.k6.io/k6/lib/netext"
+	k6types "go.k6.io/k6/lib/types"
+)
+
+func checkBlockedHosts(host string, blockedHosts *k6types.HostnameTrie) error {
+	if blockedHosts == nil {
+		return nil
+	}
+	if match, blocked := blockedHosts.Contains(host); blocked {
+		return fmt.Errorf("hostname %s is in a blocked pattern (%s)", host, match)
+	}
+	return nil
+}
+
+func checkBlockedIPs(ip net.IP, blockedIPs []*k6lib.IPNet) error {
+	for _, ipnet := range blockedIPs {
+		if ipnet.Contains(ip) {
+			// TODO: Return netext.BlackListedIPError here once its private
+			// fields are exported, or there's a constructor for it.
+			return fmt.Errorf("IP (%s) is in a blacklisted range (%s)", ip, ipnet)
+		}
+	}
+	return nil
+}
+
+// Returns a new Resolver.
+// Copied with minor changes from
+// https://github.com/grafana/k6/blob/fb70bc6f3d3f22a40e65f32deea3cea1b6d70a76/js/runner.go#L459
+func newResolver(conf k6types.DNSConfig) (k6netext.Resolver, error) {
+	ttl, err := parseTTL(conf.TTL.String)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I thought this would be good, but on second look `parseTTL` already returns a helpful error:
https://github.com/grafana/xk6-browser/blob/0cc6e6b255cdcfffb285a6301824e9a247fe58df/common/network_manager_utils.go#L72

So `cannot parse TTL` is not much more helpful.

Reply:
Yes, but `cannot parse` will allow us to follow the error chain.

Reply:
I think the error is now too noisy/repetitive, but sure, I changed it in 24b1fce.

Reply:
Feel free to use any error message as long as we can follow the error chain when an error occurs.